app/src/main/res/layout/layer_dialog.xml

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    android:gravity="center"
+    android:orientation="vertical">
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginTop="@dimen/pad_1"
+        android:text="@string/new_layer_dialog_title"
+        android:textSize="@dimen/text_1" />
+
+    <EditText
+        android:id="@+id/newLayerField"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_margin="@dimen/pad_1"
+        android:padding="10dp" />
+
+    <TextView
+        android:id="@+id/dialogCharCountError"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="@string/charCountError"
+        android:textColor="?standardErrorText"
+        android:visibility="invisible" />
+
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:gravity="center"
+        android:orientation="horizontal"
+        android:weightSum="4">
```

Comment:
Not necessary at all. Why complicate with weights?

Reply:
the weights make it fit nice and snuggly and have both buttons be equal size